src/components/ToTop.js

```diff
@@ -0,0 +1,34 @@
+import React, {useRef, useEffect, useState} from 'react';
+
+const Top = () => {
+
+  const [position, setPosition] = useState({top: 0, left: 0});
+  const scrollTop = useRef()
+
+  useEffect(() => {
+    window.scroll({
+      top: position.top,
+      left: position.left,
+      behavior: 'smooth'
+    });
+    window.addEventListener('scroll', (e) => {
+      window.scrollY > 10 
+      	? scrollTop.current.style.display = 'inline-block'
+      	: scrollTop.current.style.display = 'none'
+    });
+  })
+
+  return(
+      	<p 
+	        onClick={() => setPosition({...position, position: {top: 0, left: 0}})}               
```

Comment:
We may only need the 'top' parameter and not the left + the setPosition is just reseting the top value to 0 all the time we're pressing on it( because it's always a vertical translation and nothing else) ... so it should be simpler IMO...

```suggestion
  const [position, setPosition] = useState(0);
  const scrollTop = useRef()

  useEffect(() => {
    window.scroll({
      top: position,
      behavior: 'smooth'
    });
    window.addEventListener('scroll', (e) => {
      window.scrollY > 10 
      	? scrollTop.current.style.display = 'inline-block'
      	: scrollTop.current.style.display = 'none'
    });
  })

  return(
      	<p 
	        onClick={() => setPosition(0)}               
```

What do you think ?

PS: not tested yet !

Reply:
> Thanks for this PR, i think we can have this feature as a css stuff @ln-dev7 , but i don't ave enought skills for that , we can for now use this component if you agree to !

Good !!!
It is true that it was possible in css but he still added other featurs (like the detection of the scroll) which is only possible in JS.

I think it's good